### PR TITLE
YAZ-48-Fix-register-component

### PR DESCRIPTION
**Pull request by alejandroVaron**
What:  Fix Register Component

Why:   The registry interface is outdated

Ticket: YAZ-48

register-photo

### CODE REVIEW

File: front-end-yaz/src/components/Register.vue [hunk 1]
```diff
@@ -101,28 +108,26 @@ export default {
       valid: true,
       // Name field validations to ensure its integrity
       nameRules: [
-        (v) => !!v || "Se requiere un nombre",
+        (v) => !!v || "Se requiere un Nombre",
```
Comment (juanDuqueS) [inline front-end-yaz/src/components/Register.vue:111]: why this change?

Comment (juanDuqueS) [inline front-end-yaz/src/components/Register.vue:111]: why this change?

File: front-end-yaz/src/components/Register.vue [hunk 2]
```diff
@@ -101,28 +108,26 @@ export default {
...
         (v) =>
           (v && v.length <= 12) ||
-          "El nombre no puede pasar de los 12 caracteres",
+          "El nombre no puede exceder de los 12 caracteres",
       ],
       // Password field validations to ensure its integrity
       passwordRules: [
-        (v) => !!v || "Se requiere un nombre",
+        (v) => !!v || "Se requiere una Contraseña",
```
Comment (juanDuqueS) [inline front-end-yaz/src/components/Register.vue:118]: why the capital case?

File: front-end-yaz/src/components/Register.vue [hunk 3]
```diff
@@ -101,28 +108,26 @@ export default {
...
+          "La contraseña no puede exceder de los 24 dígitos",
       ],
-      // Validations of the Identification field to guarantee its integrity
-      idRules: [
-        (v) => !!v || "Se requiere una identificacion",
-        (v) => !isNaN(parseFloat(v)) || "Solo se permite numeros",
+      // Email field validations to ensure its integrity
+      emailRules: [
+        (v) => !!v || "Se requiere un Email",
+        (v) => /.+@.+/.test(v) || "El email no es válido",
```
Comment (juanDuqueS) [inline front-end-yaz/src/components/Register.vue:126]: you have capital case in some emails

File: front-end-yaz/src/components/Register.vue [hunk 4]
```diff
@@ -194,6 +194,32 @@ export default {
...
+    // Validations of the Identification field to guarantee its integrity
+    phoneRules() {
+      const rules = [];
+      if (this.phone != "") {
+        const rule = (v) => !isNaN(parseFloat(v)) || "Solo se permite números";
+        rules.push(rule);
+      }
+      if (this.phone != "") {
+        const rule = (v) =>
+          (v >= 0 && v <= 9999999999) ||
```
Comment (juanDuqueS) [inline front-end-yaz/src/components/Register.vue:206]: should not be better something like lenghtOf()?

File: front-end-yaz/src/components/Register.vue [hunk 5]
```diff
@@ -194,6 +194,32 @@ export default {
   computed: {
     // Declaration of Store variables
     ...mapState(["paymentProcess"]),
+    // Validations of the Identification field to guarantee its integrity
+    phoneRules() {
+      const rules = [];
+      if (this.phone != "") {
```
Comment (juanDuqueS) [inline front-end-yaz/src/components/Register.vue:200]: isEmpty() is an option?

**Review by juanDuqueS — CHANGES_REQUESTED** [review]

File: front-end-yaz/src/components/Register.vue (hunk 1 above)
Comment (alejandroVaron) [reply front-end-yaz/src/components/Register.vue]: its my mistake

File: front-end-yaz/src/components/Register.vue (hunk 2 above)
Comment (alejandroVaron) [reply front-end-yaz/src/components/Register.vue]: its my mistake, i already fix it

File: front-end-yaz/src/components/Register.vue (hunk 3 above)
Comment (alejandroVaron) [reply front-end-yaz/src/components/Register.vue:126]: its my mistake, i already fix it

File: front-end-yaz/src/components/Register.vue (hunk 4 above)
Comment (alejandroVaron) [reply front-end-yaz/src/components/Register.vue:206]: lenghtOf() is not a method for an integer

File: front-end-yaz/src/components/Register.vue (hunk 5 above)
Comment (alejandroVaron) [reply front-end-yaz/src/components/Register.vue:200]: i have to investigate more about isEmpty()

**Review by juanDuqueS — APPROVED** [review]

**Review by AngelicaAgudelo — APPROVED** [review]
Good Job 🖤
